SMLHelper/Patchers/TooltipPatcher.cs

```diff
@@ -0,0 +1,241 @@
+            {
+                // SMLHelper can access QModManager internals because of this: 
+                // https://github.com/QModManager/QModManager/blob/releases/2.0.1/QModManager/Properties/AssemblyInfo.cs#L21
+
+                string modName = null;
+
+                foreach (QMod mod in Patcher.loadedMods)
+                {
+                    if (mod == null || mod.LoadedAssembly == null) continue;
+                    if (mod.LoadedAssembly == assembly)
+                    {
+                        modName = mod.DisplayName;
+                        break;
+                    }
+                }
+
+                if (string.IsNullOrEmpty(modName))
+                {
+                    WriteModNameError(sb, "Unknown Mod", "Mod could not be determined");
+                }
+                else
+                {
+                    WriteModName(sb, modName);
+                }
+            }
+            else
+            {
+                WriteModNameError(sb, "Unknown Mod", "Assembly could not be determined");
+            }
+        }
+        internal static void WriteSpace(StringBuilder sb)
+        {
+            sb.AppendFormat("\n<size=19></size>");
+        }
+
+        internal static bool IsVanillaTechType(TechType type)
+        {
+            DisableEnumIsDefinedPatch = true;
+            bool result = Enum.IsDefined(typeof(TechType), type);
+            DisableEnumIsDefinedPatch = false;
+            return result;
+        }
+
+        #region Options
+
+        internal enum ExtraItemInfo
+        {
+            ModName,
+            ModNameAndItemID,
+            Nothing
+        }
+
+        /// <summary>
+        /// Should be one of: '<see langword="Mod name (default)">'</see>, '<see langword="Mod name and item ID"></see>', or '<see langword="Nothing"></see>'
+        /// </summary>
+        internal static ExtraItemInfo ExtraItemInfoOption { get; private set; }
+
+        internal static void SetExtraItemInfo(ExtraItemInfo value)
+        {
+            string configPath = "./QMods/Modding Helper/ExtraItemInfo.txt";
+
+            string text;
+            switch (value)
+            {
+                case ExtraItemInfo.ModName:
+                    text = "Mod name (default)";
+                    break;
+                case ExtraItemInfo.ModNameAndItemID:
+                    text = "Mod name and item ID";
+                    break;
+                case ExtraItemInfo.Nothing:
+                    text = "Nothing";
+                    break;
+                default:
+                    return;
+            }
+
+            File.WriteAllText(configPath, text);
+            ExtraItemInfoOption = value;
+        }
+
+        internal static bool Initialized = false;
+
+        internal static void Initialize()
+        {
+            if (Initialized) return;
+            Initialized = true;
+
+            string configPath = "./QMods/Modding Helper/ExtraItemInfo.txt";
```

Comment:
Since you're mentioned not liking hard-coded file paths before  
```CSharp
Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), $"ExtraItemInfo.txt");
```